src/crawlee/storages/_dataset.py

```diff
@@ -197,11 +198,27 @@ class Dataset(Storage):
     def __init__(self, id: str, name: str | None, storage_client: StorageClient) -> None:
         self._id = id
         self._name = name
+        datetime_now = datetime.now(timezone.utc)
+        self._storage_object: StorageMetadata = StorageMetadata(
+            id=id, name=name, accessed_at=datetime_now, created_at=datetime_now, modified_at=datetime_now
+        )
```

Comment:
I believe type inference is enough here, isn't it?

```suggestion
        self._storage_object = StorageMetadata(
            id=id, name=name, accessed_at=datetime_now, created_at=datetime_now, modified_at=datetime_now
        )
```